More xeno abilities usable on vehicles

TiviPlus opened the pull request:
<!-- ***STOP!***  Read this: If this is not a PR ready for review and merge or WIP, open it as a draft PR, using the arrow next to 'Create Pull Request'>

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Mechs

## Why It's Good For The Game

Mechs

## Changelog
:cl:
balance: All vehicles can now be crushered, punched by warrior, and hit by acid smoke
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

BraveMole reviewed (changes requested):
I assume acid spit works.

Here is my wish list :  
- wraith banish (I assume it works, does it bug out ?)
- Queen screech should do something imo
- is it attacked by xeno turrets
- defender and rav charge probably should do some damage ( probably already does)
- ravager ravage

TiviPlus commented:
> I assume acid spit works.
> 

Spit depends on xeno_damageable
> Here is my wish list :
> 
> * wraith banish (I assume it works, does it bug out ?)

it forcemoves the pilot out ill fix it seperately
> * Queen screech should do something imo

It's hardcoded to living mobs rn, BUT should affect the pilot
> * is it attacked by xeno turrets

Added
> * defender and rav charge probably should do some damage ( probably already does)

Looks like it will just take throw impact damage
> * ravager ravage

Currently uses a human loop, should I split it working on mech out into seperate pr?

BraveMole reviewed (approved):
Forgot that pr oups
